### PR TITLE
Enhanced logging - attempt 2

### PR DESCRIPTION
**Pull request by nealef**
Added legacy handling code for old mono_set_log_handler() users. 
Fix some message formatting errors
Fix log file destination processing

### CODE REVIEW

File: mono/utils/mono-logger.h [hunk 1]
```diff
@@ -10,18 +10,58 @@ mono_trace_set_level_string (const char *value);
...
+	void		*user_data;	/* User data from legacy handler */
+	mono_bool       header;		/* Whether we want pid/time/date in log message */
+} MonoLogCallParm;
+
+void
+mono_trace_set_log_handler_internal (MonoLogCallParm *callback, void *user_data);
 
 MONO_API void
-mono_trace_set_log_handler (MonoLogCallback callback, void *user_data);
+mono_trace_set_log_handler (MonoLogCallback callback, const char *dest, void *user_data);
```
Comment (lambdageek) [inline mono/utils/mono-logger.h]: `mono_trace_set_log_handler` shouldn't take a `dest` argument - we can't change the legacy API.

Comment (nealef) [inline mono/utils/mono-logger.h]: Oops, missed that one. Fixed.

**Comment by lambdageek** [comment]
There's something going on with the branches - this is a PR from your new-logging branch to a new-logging branch in the Mono repo.  You probably meant to open a PR against Mono master.

**Comment by nealef** [comment]
I thought I had followed the instructions properly but it appears not. I think it may be best to clone from scratch, purge my github repo, create a new branch, revert the commit that took out my original changes, reapply my latest changes, push to my github a/c, and generate a new PR. Unless you have a suggestion as to how I could get things back into shape? If I do the former can I capture the commits from today so I can just reapply them to the new branch?

**Comment by lambdageek** [comment]
I don't think you need to delete anything - the branches look ok; I think it's just a github problem... Go to https://github.com/nealef/mono/tree/new-logging and then click on "pull request" and then make sure that it's showing base fork: mono/mono, base: master, and nealef/mono as head fork and compare: new-logging as the branch.

**Comment by nealef** [comment]
No, it was showing base: new-logging. I changed it to master. 3191
generated.

**Comment by lambdageek** [comment]
Closing this PR; continuing with #3191.

**Pull request closed without merging**